magento/devdocs#: Routing. Action result object. (#4541)
magento/devdocs: guides/v2.1/extension-dev-guide/routing.md

{: .bs-callout .bs-callout-tip }
Use the [`ActionFactory`] in your router to create an instance of an `Action` class.

{: .bs-callout .bs-callout-info }
Action class should return a `result object`.

## Result object

Name | Description
--- | ---

> `json` | Sets `Content-Type:application/json` in the header and returns a json encoded representation of an array with data.

Reviewer: remove ending period

Author: **done**

> `raw` | Returns the data as it's been set. Does not set a `Content-Type` of the header.

Reviewer: ```suggestion
`raw` | Returns the data as it's been set. Does not set a `Content-Type` in the header.
```
Also, remove period from end of last sentence.

Author: **done**

> `redirect` | Creates an external redirect, which the browser follows and requests a new url.

Reviewer: remove period from end of sentence

Author: **done**

> `forward` | Internally calls the execute method of another action class and does not trigger a new request from the browser. The URL stays the same.

Reviewer: Remove period from end of last sentence.

Author: **done**

> `layout` | View result. A generic layout response can be used for rendering any kind of layout. So it comprises a response body from the layout elements it has and sets it to the HTTP response

Reviewer: ```suggestion
`layout` | View result. You can use a generic layout response to render any kind of layout. The layout comprises a response body from its layout elements and sets it to the HTTP response
```

Author: **done**

> `page` | View result. Encapsulates page type, page configuration and imposes certain layout handles. Triggers `layout.xml` rendering into HTML.

Reviewer: ```suggestion
`page` | View result. Encapsulates page type, page configuration, and imposes certain layout handles. `page` triggers `layout.xml` to render into HTML
```

Author: **done**
